src/Composer/ComposerWindow.vala

```diff
@@ -62,5 +62,11 @@ public class Mail.ComposerWindow : Hdy.ApplicationWindow {
         window_position = Gtk.WindowPosition.CENTER_ON_PARENT;
 
         add (content_grid);
+
+        key_release_event.connect ((event_key) => {
+            if (event_key.keyval == Gdk.Key.Escape) {
+                close ();
+            }
+        });
```

Comment:
Any chance we can move this to `ComposerWidget` and connect it there with the `ACTION_DISCARD` - maybe with an accelerator for `ESC` or something? This should then chain up to `composer_widget.discarded.connect (...)` in `ComposerWindow`.

The reasoning with this is, it would also allow us close the inline composer with `ESC` 🥳